From 1c001d49e9e1633f46f131782e0f824087c8aaf6 Mon Sep 17 00:00:00 2001
From: Stuart Yoder <stuart.yoder@arm.com>
Date: Mon, 16 Aug 2021 17:21:29 -0500
Subject: [PATCH 31/37] uefi-sct/SctPkg: Secure Boot: remove image load test
 assertions

Remove test assertions that test revocation of certificates based
on the certificate hash for SHA384 and SHA512.  The sbsign tools
only support generating a siglist for SHA256.

Remove image loading assertions and the corresponding image
execution info table assertions.

Remove generation of test images 7 and 8 since they are no longer
used.

Signed-off-by: Stuart Yoder <stuart.yoder@arm.com>
---
 .../Dependency/Images/GNUmakefile             |  10 +-
 .../SecureBoot/BlackBoxTest/Guid.c            |  10 --
 .../SecureBoot/BlackBoxTest/Guid.h            |  25 ----
 .../BlackBoxTest/ImageLoadingBBTest.c         | 132 ++----------------
 4 files changed, 10 insertions(+), 167 deletions(-)

diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Dependency/Images/GNUmakefile b/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Dependency/Images/GNUmakefile
index a3d65434..38274263 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Dependency/Images/GNUmakefile
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Dependency/Images/GNUmakefile
@@ -49,7 +49,7 @@ TEST_PK1_CRT=$(BIN_DIR)/SecureBoot_TestPK1.crt
 TEST_PK1_KEY=$(BIN_DIR)/SecureBoot_TestPK1.key
 endif
 
-all: $(TEST_KEYS) TestKEK2 TestKEK3 KEKSigList1.auth KEKSigList3.auth TestImage1.bin TestImage2.bin TestImage3.bin TestImage4.bin TestImage5.bin TestImage6.bin TestImage7.bin TestImage8.bin TestImage9.bin TestImage10.bin NullKEK.auth NullDB.auth NullDBX.auth TestKEK1.auth TestDB1.auth TestDBX1.auth SignCert1 dbSigList1.auth SignCert2 SignCert3 SignCert4 RevokedCert1 RevokedCert2 RevokedCert3 RevokedCert4 dbSigList2.auth dbSigList3.auth dbSigList4.auth DBXRevokedList1.auth dbSigList5.auth
+all: $(TEST_KEYS) TestKEK2 TestKEK3 KEKSigList1.auth KEKSigList3.auth TestImage1.bin TestImage2.bin TestImage3.bin TestImage4.bin TestImage5.bin TestImage6.bin TestImage9.bin TestImage10.bin NullKEK.auth NullDB.auth NullDBX.auth TestKEK1.auth TestDB1.auth TestDBX1.auth SignCert1 dbSigList1.auth SignCert2 SignCert3 SignCert4 RevokedCert1 RevokedCert2 RevokedCert3 RevokedCert4 dbSigList2.auth dbSigList3.auth dbSigList4.auth DBXRevokedList1.auth dbSigList5.auth
 
 TestImage1.bin:
 	cp $(SOURCE_DIR)/Application1.efi $(TARGET)_$(@)
@@ -73,14 +73,6 @@ TestImage6.bin: RevokedCert1
 	cp $(SOURCE_DIR)/Application2.efi $(TARGET)_$(@)
 	sbsign --key $(TARGET)_RevokedCert1.key --cert $(TARGET)_RevokedCert1.crt $(TARGET)_$(@) --output $(TARGET)_$(@)
 
-TestImage7.bin: RevokedCert2
-	cp $(SOURCE_DIR)/Application2.efi $(TARGET)_$(@)
-	sbsign --key $(TARGET)_RevokedCert2.key --cert $(TARGET)_RevokedCert2.crt $(TARGET)_$(@) --output $(TARGET)_$(@)
-
-TestImage8.bin: RevokedCert3
-	cp $(SOURCE_DIR)/Application2.efi $(TARGET)_$(@)
-	sbsign --key $(TARGET)_RevokedCert3.key --cert $(TARGET)_RevokedCert3.crt $(TARGET)_$(@) --output $(TARGET)_$(@)
-
 TestImage9.bin: RevokedCert4
 	cp $(SOURCE_DIR)/Application2.efi $(TARGET)_$(@)
 	sbsign --key $(TARGET)_RevokedCert4.key --cert $(TARGET)_RevokedCert4.crt $(TARGET)_$(@) --output $(TARGET)_$(@)
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Guid.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Guid.c
index c8c773b4..c3fceba0 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Guid.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Guid.c
@@ -92,13 +92,3 @@ EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid010 = EFI_TEST_SECUREBOOTIMAG
 EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid011 = EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_011_GUID;
 
 EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid012 = EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_012_GUID;
-
-EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid013 = EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_013_GUID;
-
-EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid014 = EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_014_GUID;
-
-EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid015 = EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_015_GUID;
-
-EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid016 = EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_016_GUID;
-
-EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid017 = EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_017_GUID;
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Guid.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Guid.h
index ebb450ae..1203c914 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Guid.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Guid.h
@@ -178,28 +178,3 @@ extern EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid011;
 { 0x00c3c2f2, 0x39d5, 0x4d35, {0xb7, 0xe7, 0x58, 0x7c, 0xa0, 0xf3, 0xcb, 0x75}}
 
 extern EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid012;
-
-#define EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_013_GUID \
-{ 0x55211c3f, 0xc438, 0x485a, {0xb1, 0x75, 0xb1, 0xef, 0x5d, 0x31, 0x41, 0xc7}}
-
-extern EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid013;
-
-#define EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_014_GUID \
-{ 0x32a28ae3, 0x9dcb, 0x4f3e, {0x86, 0x24, 0xe3, 0x3e, 0x50, 0xa9, 0x33, 0xe1}}
-
-extern EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid014;
-
-#define EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_015_GUID \
-{ 0x9031ee38, 0x3e1a, 0x44f2, {0x8e, 0x08, 0xf5, 0x2b, 0x52, 0xad, 0x1b, 0xd9}}
-
-extern EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid015;
-
-#define EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_016_GUID \
-{ 0x73aae70e, 0x7953, 0x4983, {0x89, 0x4a, 0x1e, 0x6a, 0xf2, 0x78, 0x25, 0xd7}}
-
-extern EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid016;
-
-#define EFI_TEST_SECUREBOOTIMAGELOADING_ASSERTION_017_GUID \
-{ 0x26d2b894, 0x37c3, 0x44d5, {0x8b, 0xf3, 0x6c, 0xa6, 0x01, 0x1a, 0xb4, 0x62}}
-
-extern EFI_GUID gSecureBootImageLoadingBbTestAssertionGuid017;
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/ImageLoadingBBTest.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/ImageLoadingBBTest.c
index 2bde390b..8fda7763 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/ImageLoadingBBTest.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/ImageLoadingBBTest.c
@@ -787,87 +787,7 @@ ImageLoadingTestCheckpoint2 (
                  );
 
   //
-  //  Test assertion 2: Verify revocation with signed image with signature in db but
-  //  revoked in dbx with EFI_CERT_X509_SHA384_GUID.  Load image (TestImage7) signed
-  //  with RevokedCert2.  Expect result: SECURITY_VIOLATION.
-  //
-
-  FileName = L"TestImage7.bin";
-  EntireFileName = SctPoolPrint (L"%s\\%s", gFilePath, FileName);
-  FilePath = SctFileDevicePath (gDeviceHandle, EntireFileName);
-  SctFreePool (EntireFileName);
-
-  ImageHandle = NULL;
-  Status = gtBS->LoadImage (
-                     FALSE,
-                     gImageHandle,
-                     FilePath,
-                     NULL,
-                     0,
-                     &ImageHandle
-                     );
-
-  if (Status == EFI_SECURITY_VIOLATION) {
-    Result = EFI_TEST_ASSERTION_PASSED;
-  } else if (Status == EFI_ACCESS_DENIED && ImageHandle == NULL) {
-    Result = EFI_TEST_ASSERTION_PASSED;
-  } else {
-    Result = EFI_TEST_ASSERTION_FAILED;
-  }
-
-  StandardLib->RecordAssertion (
-                 StandardLib,
-                 Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid005,
-                 L"SecureBoot - Verify load of signed image with signature in db and revoked in dbx with SHA384 hash.",
-                 L"%a:%d:Status - %r",
-                 __FILE__,
-                 (UINTN)__LINE__,
-                 Status
-                 );
-
-  //
-  //  Test assertion 3: Verify revocation with signed image (TestImage8) with signature in
-  //  db but revoked in dbx with EFI_CERT_X509_SHA512_GUID.  Load image signed with
-  //  RevokedCert3.  Expect result: SECURITY_VIOLATION.
-  //
-
-  FileName = L"TestImage8.bin";
-  EntireFileName = SctPoolPrint (L"%s\\%s", gFilePath, FileName);
-  FilePath = SctFileDevicePath (gDeviceHandle, EntireFileName);
-  SctFreePool (EntireFileName);
-
-  ImageHandle = NULL;
-  Status = gtBS->LoadImage (
-                     FALSE,
-                     gImageHandle,
-                     FilePath,
-                     NULL,
-                     0,
-                     &ImageHandle
-                     );
-
-  if (Status == EFI_SECURITY_VIOLATION) {
-    Result = EFI_TEST_ASSERTION_PASSED;
-  } else if (Status == EFI_ACCESS_DENIED && ImageHandle == NULL) {
-    Result = EFI_TEST_ASSERTION_PASSED;
-  } else {
-    Result = EFI_TEST_ASSERTION_FAILED;
-  }
-
-  StandardLib->RecordAssertion (
-                 StandardLib,
-                 Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid005,
-                 L"SecureBoot - Verify load of signed image with signature in db and revoked in dbx with SHA512 hash.",
-                 L"%a:%d:Status - %r",
-                 __FILE__,
-                 (UINTN)__LINE__,
-                 Status
-                 );
-
-  //
-  //  Test assertion 4: Verify revocation with image (TestImage9) with signature in db,
+  //  Test assertion 2: Verify revocation with image (TestImage9) with signature in db,
   //  but signature also in DBXRevokedList1.  Load image signed with RevokedCert4.
   //  Expect result: SECURITY_VIOLATION.
   //
@@ -898,7 +818,7 @@ ImageLoadingTestCheckpoint2 (
   StandardLib->RecordAssertion (
                  StandardLib,
                  Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid005,
+                 gSecureBootImageLoadingBbTestAssertionGuid006,
                  L"SecureBoot - Verify load of signed image with signature in db and revoked in dbx with certificate.",
                  L"%a:%d:Status - %r",
                  __FILE__,
@@ -907,7 +827,7 @@ ImageLoadingTestCheckpoint2 (
                  );
 
   //
-  //  Test assertion 5: Verify revocation with image with signature in db,
+  //  Test assertion 3: Verify revocation with image with signature in db,
   //  but revoked with SHA256 hash of image in DBXRevokedList1.  Load image
   //  (TestImage10).  Expect result: SECURITY_VIOLATION.
   //
@@ -938,7 +858,7 @@ ImageLoadingTestCheckpoint2 (
   StandardLib->RecordAssertion (
                  StandardLib,
                  Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid005,
+                 gSecureBootImageLoadingBbTestAssertionGuid007,
                  L"SecureBoot - Verify load of signed image with signature in db and revoked in dbx with hash of image.",
                  L"%a:%d:Status - %r",
                  __FILE__,
@@ -1075,7 +995,7 @@ ImageLoadingTestCheckpoint3 (
   StandardLib->RecordAssertion (
                  StandardLib,
                  Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid011,
+                 gSecureBootImageLoadingBbTestAssertionGuid008,
                  L"SecureBoot - Verify load of TestImage1.bin recorded in Image Execution Info Table.",
                  L"%a:%d:Status - %r",
                  __FILE__,
@@ -1092,7 +1012,7 @@ ImageLoadingTestCheckpoint3 (
   StandardLib->RecordAssertion (
                  StandardLib,
                  Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid012,
+                 gSecureBootImageLoadingBbTestAssertionGuid009,
                  L"SecureBoot - Verify load of TestImage2.bin recorded in Image Execution Info Table.",
                  L"%a:%d:Status - %r",
                  __FILE__,
@@ -1109,7 +1029,7 @@ ImageLoadingTestCheckpoint3 (
   StandardLib->RecordAssertion (
                  StandardLib,
                  Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid013,
+                 gSecureBootImageLoadingBbTestAssertionGuid010,
                  L"SecureBoot - Verify load of TestImage6.bin recorded in Image Execution Info Table.",
                  L"%a:%d:Status - %r",
                  __FILE__,
@@ -1117,40 +1037,6 @@ ImageLoadingTestCheckpoint3 (
                  Status
                  );
 
-  if (VerifyImageEntry(L"TestImage7.bin", EFI_IMAGE_EXECUTION_AUTH_SIG_FAILED)) {
-    Result = EFI_TEST_ASSERTION_FAILED;
-  } else {
-    Result = EFI_TEST_ASSERTION_PASSED;
-  }
-
-  StandardLib->RecordAssertion (
-                 StandardLib,
-                 Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid014,
-                 L"SecureBoot - Verify load of TestImage7.bin recorded in Image Execution Info Table.",
-                 L"%a:%d:Status - %r",
-                 __FILE__,
-                 (UINTN)__LINE__,
-                 Status
-                 );
-
-  if (VerifyImageEntry(L"TestImage8.bin", EFI_IMAGE_EXECUTION_AUTH_SIG_FAILED)) {
-    Result = EFI_TEST_ASSERTION_FAILED;
-  } else {
-    Result = EFI_TEST_ASSERTION_PASSED;
-  }
-
-  StandardLib->RecordAssertion (
-                 StandardLib,
-                 Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid015,
-                 L"SecureBoot - Verify load of TestImage8.bin recorded in Image Execution Info Table.",
-                 L"%a:%d:Status - %r",
-                 __FILE__,
-                 (UINTN)__LINE__,
-                 Status
-                 );
-
   if (VerifyImageEntry(L"TestImage9.bin", EFI_IMAGE_EXECUTION_AUTH_SIG_FAILED)) {
     Result = EFI_TEST_ASSERTION_FAILED;
   } else {
@@ -1160,7 +1046,7 @@ ImageLoadingTestCheckpoint3 (
   StandardLib->RecordAssertion (
                  StandardLib,
                  Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid016,
+                 gSecureBootImageLoadingBbTestAssertionGuid011,
                  L"SecureBoot - Verify load of TestImage9.bin recorded in Image Execution Info Table.",
                  L"%a:%d:Status - %r",
                  __FILE__,
@@ -1177,7 +1063,7 @@ ImageLoadingTestCheckpoint3 (
   StandardLib->RecordAssertion (
                  StandardLib,
                  Result,
-                 gSecureBootImageLoadingBbTestAssertionGuid017,
+                 gSecureBootImageLoadingBbTestAssertionGuid012,
                  L"SecureBoot - Verify load of TestImage10.bin recorded in Image Execution Info Table.",
                  L"%a:%d:Status - %r",
                  __FILE__,
-- 
2.17.1

